compiler/src/iree/compiler/Dialect/HAL/IR/HALOpFolders.cpp

```diff
@@ -165,17 +165,24 @@ void BufferViewCreateOp::getCanonicalizationPatterns(RewritePatternSet &results,
 namespace {
 
 /// Skips a hal.buffer_view.buffer accessor when the buffer view was created in
-/// the same scope and we know the origin buffer.
+/// the same scope at zero offset and we know the origin buffer.
 struct SkipBufferViewBufferOp : public OpRewritePattern<BufferViewBufferOp> {
   using OpRewritePattern<BufferViewBufferOp>::OpRewritePattern;
 
   LogicalResult matchAndRewrite(BufferViewBufferOp op,
                                 PatternRewriter &rewriter) const override {
-    if (auto createOp = dyn_cast_or_null<BufferViewCreateOp>(
-            op.getBufferView().getDefiningOp())) {
+    auto createOp = dyn_cast_or_null<BufferViewCreateOp>(
+        op.getBufferView().getDefiningOp());
+    if (!createOp)
+      return failure();
+
+    APInt offset;
+    if (matchPattern(createOp.getSourceOffset(), m_ConstantInt(&offset)) &&
```

Comment:
you could just replace this with `m_Zero()`?

Reply:
Thanks, didn't know it exist